acme/api/order.go

```diff
@@ -280,44 +337,28 @@ func newAuthorization(ctx context.Context, az *acme.Authorization) error {
 		var target string
 		switch az.Identifier.Type {
 		case acme.WireUser:
-			wireOptions, err := prov.GetOptions().GetWireOptions()
-			if err != nil {
-				return acme.WrapErrorISE(err, "failed getting Wire options")
+			wireOptions := prov.GetOptions().GetWireOptions()
+			if wireOptions == nil {
+				return acme.NewErrorISE("failed getting Wire options")
 			}
-			var targetProvider interface{ EvaluateTarget(string) (string, error) }
-			switch typ {
-			case acme.WIREOIDC01:
-				targetProvider = wireOptions.GetOIDCOptions()
-			default:
-				return acme.NewError(acme.ErrorMalformedType, "unsupported type %q", typ)
-			}
-
-			target, err = targetProvider.EvaluateTarget("")
+			target, err = wireOptions.GetOIDCOptions().EvaluateTarget("")
```

Comment:
Do we ever pass the deviceID to these options?

Reply:
Need to check this one with Wire, as it may no longer be used after changing to two different identifier types. But maybe they still need it with a UserID instead.